### PR TITLE
CDK: Add jinja macro `format_datetime`

### LINKED ISSUE #20292: CDK: Add jinja macro format_datetime
## What
In `source-chartmogul` I want to do next:
```
start-date: "{{ format_datetime(config.start_date, "%Y-%m-%d") }}"
```

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
In `source-chartmogul` I want to do next:
```
start-date: "{{ format_datetime(config.start_date, "%Y-%m-%d") }}"
```

### CODE REVIEW

**Review by sherifnada — CHANGES_REQUESTED** [review]
@grubberr thanks for the contribution! I have two request for changes: 

1. `strptime` is not an easily understood/remembered name especially if we want non-engineers to use the macro.
2. Could you update the docstring to indicate where to find formats, and some examples to understand how it would work?

**Comment by grubberr** [comment]
@sherifnada @brianjlai @girarda  can you please review?
I would like to use it inside Low-code connector `source-chartmogul`

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py [hunk 1]
```diff
@@ -96,5 +96,17 @@ def day_delta(num_days: int) -> str:
     return (datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=num_days)).strftime("%Y-%m-%dT%H:%M:%S.%f%z")
 
 
-_macros_list = [now_local, now_utc, today_utc, timestamp, max, day_delta]
+def datetime_to_format(dt: Union[str, datetime.datetime], format: str):
```
Comment (sherifnada) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py:99]: can we rename to `format_datetime`?

**Review by sherifnada — APPROVED** [review]

**Pull request merged**